### PR TITLE
Fix pending checkpoints submission

### PR DESCRIPTION
**Pull request by Stefan-Ethernal**
# Description

This PR fixes epoch ending blocks detection for submitting pending checkpoints. Prior to this change, for first pending checkpoint block, we were checkpointing epoch-beginning instead of epoch-ending block.

# Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### CODE REVIEW

**Review by goran-ethernal — APPROVED** [review]

**Comment by ferranbt** [comment]
Can we include any unit test or e2e test to prevent this?

**Comment by Stefan-Ethernal** [comment]
> Can we include any unit test or e2e test to prevent this?

Sure I meant to implement a unit test, although haven't yet made it. Stay tuned :slightly_smiling_face:

**Comment by ferranbt** [comment]
> Sure I meant to implement a unit test, although haven't yet made it. Stay tuned

Sorry I thought it was ready for review!

**Review by b00ris — APPROVED** [review]

File: consensus/polybft/checkpoint_manager_test.go [hunk 1]
```diff
@@ -274,25 +268,52 @@ var _ txrelayer.TxRelayer = (*dummyTxRelayer)(nil)
 
 type dummyTxRelayer struct {
 	mock.Mock
+
+	test             *testing.T
```
Comment (JekaMas) [inline consensus/polybft/checkpoint_manager_test.go:272]: I am not sure, where we are using this field?

Comment (Stefan-Ethernal) [reply consensus/polybft/checkpoint_manager_test.go:272]: Here (there is a test helper function which extracts block number out of abi encoded input parameters): https://github.com/0xPolygon/polygon-edge/pull/992/files#diff-48b89eabd155ec467879f448386c719753460b4539d1dd8076f539e0d955c75fR289

**Review by Nemanja0x — APPROVED** [review]

**Pull request merged**